blog-posts/gatsby-ssr-caching.md

```diff
@@ -0,0 +1,98 @@
+- `s-maxage`: The time in seconds for how long the page should be cached by the CDN. If not set, the CDN will use `max-age`. Use this to determine how long a page can be cached globally. You may want to set the time lower for new and frequently changing content and higher for old or less frequently changing content.
+- `stale-while-revalidate` : The time in seconds for how long the page should be stored by the CDN and a new page should be requested in the background when a visitor hits it. Once the new page is requests, the stale page will be replaced by it. Use this to serve an old version of a page before the new page has been rendered. This is very powerful to keep your page fast while updating it in the background. If the content is not crucial, you can set this to a very high value.
+
+With those 3 directives, you can make your page as fast as with SSG or DSG. If the values are set correctly, only the 
+very first user after a deployment will have to wait for the rendering. Everyone else will see a lightning fast page.
+
+> This header can also be used on any other SSR framework and CDN. It's not limited to Gatsby or Gatsby Cloud.
+
+## Enable Caching
+
+Now let's dive into the code and set the `Cache-Control` header:
+
+```jsx
+// pages/ssr.jsx
+export default function Cache() {
+  return (
+      <h1>SSR: Server Side Rendering</h1>
+  )
+}
+
+export async function getServerData() {
+  return {
+    headers: {
+      'Cache-Control': 'public, max-age=10, s-maxage=60, stale-while-revalidate=240',
+    }
+  }
+}
+```
+
+Gatsby SSR pages export the `getServerData()` function, which you can use to do all kinds of stuff with it (like 
+fetching external data). Besides the props that can be injected into the page itself, this function returns a `headers` 
+object. This object is used to set the `Cache-Control` header.
+
+In our case, we set the browser cache (`max-age`) to 10 seconds and the CDN cache (`s-max-age`) to 1 minute. After 1 minute, the CDN will 
+serve stale content (`stale-while-revalidate`) for a maximum of 4 additional minutes. If there is one request within these 4 minutes, a 
+new page will be rendered in the background while the old page gets served. After the new page is rendered, it will replace 
+the old one on the CDN and the caching time starts again. Let's check the network tab to see if it works:
+
+![Chrome network tab shows that with proper caching headers only the very first request is slow](images/gatsby-ssr-caching-network-tab.png)
+
+You can see: The very first page took 5 seconds to load (for this example, I added an artificial delay of 5 seconds to make the server rendered 
+page more obvious). All other requests were served from CDN within 30ms. That's super fast! Also, after the stale page 
+was served, the page was re-rendered in the background and replaced. The request after the stale one was super fast 
+again but served the new page.
+
+## Gatsby SSR caveats
```

Comment:
Not 100% sure, but is this article about Gatsby SSR or caching?

Reply:
I changed the wording a bit. It's things to keep in mind if you want to put SSR with caching in production and it felt important to me to mention it. I agree that it doesn't fit perfectly into the article.